src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
@pranavkm while the `CookieRequestCultureProvider` needs two constructors, we need somehow to get the `HttpContext` to access the `ILogger` from the DI container
```csharp
public RequestLocalizationOptions(IHttpContextAccessor httpContextAccessor)
{
    var logger = (ILogger)httpContextAccessor.HttpContext
        .RequestServices.GetService(typeof(ILogger));

    RequestCultureProviders = new List<IRequestCultureProvider>
    {
        new QueryStringRequestCultureProvider { Options = this },
        (logger == null ?
            new CookieRequestCultureProvider { Options = this } :
            new CookieRequestCultureProvider(logger) { Options = this }),
        new AcceptLanguageHeaderRequestCultureProvider { Options = this }
    };
}
```
If I made something like the above code snippet, there will be a breaking change, which is not preferred, could you please show me a code snippet for the things in your mind.